src/cogent3/core/sequence.py

```diff
@@ -2946,51 +2946,51 @@ class ArrayProteinWithStopSequence(ArraySequence):
 
 
 @singledispatch
-def _coerce_seq(data, preserve_case, checker):
+def _coerce_to_seqview(data, seqid, preserve_case, checker):
     from cogent3.core.alignment import Aligned
 
     if isinstance(data, Aligned):
-        return _coerce_seq(str(data), preserve_case, checker)
+        return _coerce_to_seqview(data.data, seqid, preserve_case, checker)
```

Comment:
whoops, I think this is wrong because the Aligned can contain gaps which means my suggestion to directly use `.data` (which is the ungapped sequence) is a bad idea. So change this to
```python
_coerce_to_seqview(str(data), seqid, preserve_case, checker)
```